File: Iteration1-Java/src/main/java/cse3063group3/LectureHour.java
package cse3063group3;

public class LectureHour {
    private int lectureDay;
    private int lectureHourNumber;
    private Classroom classroom;

    public LectureHour(int lectureDay,int lectureHourNumber,Classroom classroom){
        this.lectureDay = lectureDay;
        this.lectureHourNumber=lectureHourNumber;
        this.classroom=classroom;
    }
    public LectureHour(int lectureDay,int lectureHourNumber){
        this.lectureDay = lectureDay;
        this.lectureHourNumber=lectureHourNumber;
    }

    public int getLectureDay(){
        return this.lectureDay;
    }

    public void setLectureDay(int lectureDay){
        this.lectureDay = lectureDay;
    }
  
    public int getLectureHourNumber(){
        return this.lectureHourNumber;
    }

    public void setLectureHourNumber(int lectureHourNumber){
        this.lectureHourNumber=lectureHourNumber;
    }

    public Classroom getClassroom(){
        return this.classroom;
    }
  
    
    public void setClassroom(Classroom classroom){
        this.classroom = classroom;
    }
    
    public int getLectureTimeTableNumber(){
        return this.lectureDay * 10 +this.lectureHourNumber;
    }
}

File: Iteration2-Python/src/LectureHour.py

from Classroom import Classroom

import logging
logger = logging.getLogger(__name__)    # to ensure that the logger is named after the module
class LectureHour():
    
    def __init__(self, lectureDay, lectureHourNumber, classroom):
        self.__lecture_day : int = lectureDay
        self.__lecture_hour_number : int = lectureHourNumber
        self.__classroom  :Classroom = classroom

    def  get_lecture_day(self):
        return self.__lecture_day
    
    def set_lecture_day(self,lectureDay):
        self.__lecture_day = lectureDay
        
    def get_lecture_hour_number(self):
        return self.__lecture_hour_number
    
    def set_lecture_hour_number(self,lectureHourNumber):
        self.__lecture_hour_number = lectureHourNumber
        
    def get_classroom(self):
        return self.__classroom

    def get_classroom_id(self):
        return self.__classroom.get_classroom_id()
    
    def set_classroom(self,classroom):
        if type(classroom)is str:
            raise TypeError
        self.__classroom = classroom
        
    def get_lecture_time_table_number(self):
        return self.__lecture_day * 10 +self.__lecture_hour_number
